### ISSUE TITLE
figure out which CDTs we can drop for cos8, maybe drop them all?

### ISSUE THREAD

**Issue by beckermr** [conda-forge/cdt-builds#66]
We need to figure out which ones we need to keep vs ones we can build?

**Comment by h-vetinari** [comment]
@conda-forge/core, I started with creating a list (based on the README, modulo aarches) and collecting some information (mainly based on the excellent pkgs.org).

If you think additional information would be useful (and don't care if the table formatting looks bad), please tell me what else I should add or remove. If/once people agree on a format I can fill in the rest.

**Comment by h-vetinari** [comment]
## X11

| Name<br/>(cos6/cos7) | Name<br/>(alma8) | Version<br/>(cos7) | Version<br/>(alma8) | Feedstock? | Comment | To replace<br/>✅  / ❌ / ❔ |
|-|-|-|-|-|-|-|
| libice | | | | | | |
| libice-devel | | | | | | |
| libsm | | | | | | |
| libsm-devel | | | | | | |
| libx11 | | | | | | |
| libx11-common | | | | | | |
| libx11-devel | | | | | | |
| libxau | | | | | | |
| libxau-devel | | | | | | |
| libxcb | | | | | | |
| libxcb-devel | | | | | | |
| libxcomposite | | | | | | |
| libxcomposite-devel | | | | | | |
| libxcursor | | | | | | |
| libxcursor-devel | | | | | | |
| libxdamage | | | | | | |
| libxdamage-devel | | | | | | |
| libxext | | | | | | |
| libxext-devel | | | | | | |
| libxfixes | | | | | | |
| libxfixes-devel | | | | | | |
| libxft | | | | | | |
| libxft-devel | | | | | | |
| libxi | | | | | | |
| libxi-devel | | | | | | |
| libxinerama | | | | | | |
| libxinerama-devel | | | | | | |
| libxkbcommon | | | | | | |
| libxkbcommon-devel | | | | | | |
| libxrandr | | | | | | |
| libxrandr-devel | | | | | | |
| libxrender | | | | | | |
| libxrender-devel | | | | | | |
| libxscrnsaver | | | | | | |
| libxscrnsaver-devel | | | | | | |
| libxshmfence | | | | | | |
| libxshmfence-devel | | | | | | |
| libxt | | | | | | |
| libxt-devel | | | | | | |
| libxtst | | | | | | |
| libxtst-devel | | | | | | |
| libxxf86vm | | | | | | |
| libxxf86vm-devel | | | | | | |
| xcb-util | | | | | | |
| xcb-util-devel | | | | | | |
| xcb-util-image | | | | | | |
| xcb-util-image-devel | | | | | | |
| xcb-util-keysyms | | | | | | |
| xcb-util-keysyms-devel | | | | | | |
| xcb-util-renderutil | | | | | | |
| xcb-util-renderutil-devel | | | | | | |
| xcb-util-wm | | | | | | |
| xcb-util-wm-devel | | | | | | |
| xorg-x11-proto-devel | | | | | | |
| xorg-x11-server-common | | | | | | |
| xorg-x11-server-xvfb | | | | | | |
| xorg-x11-util-macros | | | | | | |

## GNOME

| Name<br/>(cos6/cos7) | Name<br/>(alma8) | Version<br/>(cos7) | Version<br/>(alma8) | Feedstock? | Comment | To replace<br/>✅  / ❌ / ❔ |
|-|-|-|-|-|-|-|
| [at-spi](https://centos.pkgs.org/7/centos-x86_64/at-spi-1.32.0-12.el7.x86_64.rpm.html) | [at-spi2-core](https://almalinux.pkgs.org/8/almalinux-appstream-x86_64/at-spi2-core-2.28.0-1.el8.x86_64.rpm.html)(?) | 1.32.0 | 2.28.0 | | Libname changed<br/>`libspi.so`->`libatspi.so` | |
| at-spi-devel | [at-spi2-core-devel](https://almalinux.pkgs.org/8/almalinux-appstream-x86_64/at-spi2-core-devel-2.28.0-1.el8.x86_64.rpm.html)(?) | 1.32.0 | 2.28.0 | | | |
| [atk](https://centos.pkgs.org/7/centos-x86_64/atk-2.28.1-2.el7.x86_64.rpm.html) | [same](https://almalinux.pkgs.org/8/almalinux-appstream-x86_64/atk-2.28.1-1.el8.x86_64.rpm.html) | 2.28.1 | 2.28.1 | | | |
| atk-devel | same | 2.28.1 | 2.28.1 | | | |
| gconf2 | [same](https://almalinux.pkgs.org/8/almalinux-appstream-x86_64/GConf2-3.2.6-22.el8.x86_64.rpm.html) | 3.2.6 | 3.2.6 | | | |
| gdk-pixbuf2 | [same](https://almalinux.pkgs.org/8/almalinux-baseos-x86_64/gdk-pixbuf2-2.36.12-5.el8.x86_64.rpm.html) | 2.36.12 | 2.36.12 | | | |
| gdk-pixbuf2-devel | same | 2.36.12 | 2.36.12 | | | |
| glib-networking | [same](https://almalinux.pkgs.org/8/almalinux-baseos-x86_64/glib-networking-2.56.1-1.1.el8.x86_64.rpm.html) | 2.56.1 | 2.56.1 | | | |
| glib2 | [same](https://almalinux.pkgs.org/8/almalinux-baseos-x86_64/glib2-2.56.4-161.el8.x86_64.rpm.html) | 2.56.1 | 2.56.4 | | | |
| glib2-devel | same | 2.56.1 | 2.56.4 | | | | 
| gtk2 | [same](https://almalinux.pkgs.org/8/almalinux-appstream-x86_64/gtk2-2.24.32-5.el8.x86_64.rpm.html) | 2.24.31 | 2.24.32 | | | |
| gtk2-devel | same | 2.24.31 | 2.24.32 | | | | 
| gtkmm24 | [same](https://almalinux.pkgs.org/8/almalinux-appstream-x86_64/gtkmm24-2.24.5-6.el8.x86_64.rpm.html) | 2.24.5 | 2.24.5 | | | |
| gtkmm24-devel | same | 2.24.5 | 2.24.5 | | | | 
| libbonobo | | | | | | |
| libbonobo-devel | | | | | | |
| libidl | | | | | | |
| libidl-devel | | | | | | |
| orbit2 | | | | | | |
| orbit2-devel | | | | | | |
| webkitgtk | | | | | | |
| webkitgtk-devel | | | | | | |

## Other Graphics (DRM, Mesa, OpenGL, etc.)

| Name<br/>(cos6/cos7) | Name<br/>(alma8) | Version<br/>(cos7) | Version<br/>(alma8) | Feedstock? | Comment | To replace<br/>✅  / ❌ / ❔ |
|-|-|-|-|-|-|-|
| cairo | [same](https://almalinux.pkgs.org/8/almalinux-appstream-x86_64/cairo-1.15.12-6.el8.x86_64.rpm.html) | 1.15.12 | 1.15.12 | [yes](https://github.com/conda-forge/cairo-feedstock)| | |
| cairo-devel | same | 1.15.12 | 1.15.12 | | | |
| libdrm | | | | | | |
| libdrm-devel | | | | | | |
| libglvnd | | | | | | |
| libglvnd-core-devel | | | | | | |
| libglvnd-devel | | | | | | |
| libglvnd-egl | | | | | | |
| libglvnd-gles | | | | | | |
| libglvnd-glx | | | | | | |
| libglvnd-opengl | | | | | | |
| mesa-dri-drivers | | | | | | |
| mesa-dri1-drivers | | | | | | |
| mesa-khr-devel | | | | | | |
| mesa-libegl | | | | | | |
| mesa-libegl-devel | | | | | | |
| mesa-libgbm | | | | | | |
| mesa-libgl | | | | | | |
| mesa-libgl-devel | | | | | | |
| mesa-libglapi | | | | | | |
| pixman | | | | | | |
| pixman-devel | | | | | | |

## Other Hardware & Kernel Interfaces

| Name<br/>(cos6/cos7) | Name<br/>(alma8) | Version<br/>(cos7) | Version<br/>(alma8) | Feedstock? | Comment | To replace<br/>✅  / ❌ / ❔ |
|-|-|-|-|-|-|-|
| alsa-lib | | | | [yes](https://github.com/conda-forge/alsa-lib-feedstock) | Sound | |
| alsa-lib-devel | | | | | | |
| cups-devel | same | 1.6.3 | 2.2.6 | | | |
| cups-libs | [same](https://almalinux.pkgs.org/8/almalinux-baseos-x86_64/cups-libs-2.2.6-54.el8_9.x86_64.rpm.html) | 1.6.3 | 2.2.6 | | Printer | |
| ibacm | | | | | Infiniband | |
| libibcm | | | | | | |
| libibcm-devel | | | | | | |
| libibmad | | | | | | |
| libibmad-devel | | | | | | |
| libibumad | | | | | | |
| libibumad-devel | | | | | | |
| libibverbs | | | | | | |
| libibverbs-devel | | | | | | |
| libnl | | | | | Kernel Netlib Sockets | |
| libnl-devel | | | | | | |
| libnl3 | | | | | | |
| libnl3-cli | | | | | | |
| libnl3-devel | | | | | | |
| librdmacm | | | | | [RDMA](https://en.wikipedia.org/wiki/Remote_direct_memory_access) | |
| librdmacm-devel | | | | | | |
| numactl | | | | | | |
| numactl-devel | | | | | | |
| numactl-libs | | | | | | |
| opensm | | | | | | |
| opensm-devel | | | | | | |
| opensm-libs | | | | | | |
| pciutils | | | | | | |
| pciutils-devel | | | | | | |
| pciutils-libs | | | | | | |
| rdma-core | | | | | | |
| rdma-core-devel | | | | | | |

## System & Security Libs

| Name<br/>(cos6/cos7) | Name<br/>(alma8) | Version<br/>(cos7) | Version<br/>(alma8) | Feedstock? | Comment | To replace<br/>✅  / ❌ / ❔ |
|-|-|-|-|-|-|-|
| centos-release | | | | | No direct match? | |
| chkconfig | [same](https://almalinux.pkgs.org/8/almalinux-baseos-x86_64/chkconfig-1.19.2-1.el8.x86_64.rpm.html) | 1.7.6 | 1.19.2 | | | |
| kernel-headers | | | | | | |
| kmod | | | | | | |
| libselinux | | | | | | |
| libselinux-devel | | | | | | |
| libsepol | | | | | | |
| libsepol-devel | | | | | | |
| libudev | | | | | not found on pkgs.org for cos7? | |
| libudev-devel | | | | | | |
| p11-kit | | | | | | |
| p11-kit-trust | | | | | | |
| systemd | | | | | | |
| systemd-devel | | | | | | |
| systemd-libs | | | | | | |

## Formats, Parsing & Images

| Name<br/>(cos6/cos7) | Name<br/>(alma8) | Version<br/>(cos7) | Version<br/>(alma8) | Feedstock? | Comment | To replace<br/>✅  / ❌ / ❔ |
|-|-|-|-|-|-|-|
| expat | [same](https://almalinux.pkgs.org/8/almalinux-baseos-x86_64/expat-2.2.5-11.el8.x86_64.rpm.html) | 2.1.0 | 2.2.5 | [yes](https://github.com/conda-forge/expat-feedstock) | | |
| expat-devel | same | 2.1.0 | 2.2.5 | | | |
| libjpeg-turbo | same | 1.2.90 | 1.5.3 | [yes](https://github.com/conda-forge/libjpeg-turbo-feedstock) | | |
| libpng | same | 1.5.13 | 1.6.34 | [yes](https://github.com/conda-forge/libpng-feedstock) | Upstream provides several versions | |
| libpng-devel | same | 1.5.13 | 1.6.34 | | | |
| libtiff | | | | [yes](https://github.com/conda-forge/libtiff-feedstock) | | |
| libuuid | | | | [yes](https://github.com/conda-forge/libuuid-feedstock) | | |
| libuuid-devel | | | | | | |
| libxml2 | | | | [yes](https://github.com/conda-forge/libxml2-feedstock) | | |
| libxml2-devel | | | | | | |
| pcre | | | | [yes](https://github.com/conda-forge/pcre-feedstock) | | |

## Fonts & Language

| Name<br/>(cos6/cos7) | Name<br/>(alma8) | Version<br/>(cos7) | Version<br/>(alma8) | Feedstock? | Comment | To replace<br/>✅  / ❌ / ❔ |
|-|-|-|-|-|-|-|
| fontconfig | [same](https://almalinux.pkgs.org/8/almalinux-baseos-x86_64/fontconfig-2.13.1-4.el8.x86_64.rpm.html) | 2.13.0 | 2.13.1 | [yes](https://github.com/conda-forge/fontconfig-feedstock) | | |
| fontconfig-devel | same | 2.13.0 | 2.13.1 | | | |
| freetype | [same](https://almalinux.pkgs.org/8/almalinux-baseos-x86_64/freetype-2.9.1-9.el8.x86_64.rpm.html) | 2.8 | 2.9.1 | [yes](https://github.com/conda-forge/freetype-feedstock) | | |
| freetype-devel | same | 2.8 | 2.9.1 | | | |
| fribidi | same | 1.0.2 | 1.0.4 | [yes](https://github.com/conda-forge/fribidi-feedstock) | | |
| harfbuzz | | | | [yes](https://github.com/conda-forge/harfbuzz-feedstock) | | |
| libthai | | | | | | |
| pango | | | | [yes](https://github.com/conda-forge/pango-feedstock) | | |
| pango-devel | | | | | | |

## Web & Browser

| Name<br/>(cos6/cos7) | Name<br/>(alma8) | Version<br/>(cos7) | Version<br/>(alma8) | Feedstock? | Comment | To replace<br/>✅  / ❌ / ❔ |
|-|-|-|-|-|-|-|
| ca-certificates | [same](https://almalinux.pkgs.org/8/almalinux-baseos-x86_64/ca-certificates-2023.2.60_v7.0.306-80.0.el8_8.noarch.rpm.html) | | | [yes](https://github.com/conda-forge/ca-certificates-feedstock) | | |
| libsoup | | | | [yes](https://github.com/conda-forge/libsoup-feedstock) | HTTP | |
| libsoup-devel | | | | | | |
| nspr | | | | [yes](https://github.com/conda-forge/nspr-feedstock) | | |
| nss | | | | [yes](https://github.com/conda-forge/nss-feedstock) | | |
| nss-softokn | | | | | | |
| nss-softokn-freebl | | | | | | |
| nss-util | | | | | | |

## Rest

| Name<br/>(cos6/cos7) | Name<br/>(alma8) | Version<br/>(cos7) | Version<br/>(alma8) | Feedstock? | Comment | To replace<br/>✅  / ❌ / ❔ |
|-|-|-|-|-|-|-|
| copy-jdk-configs | [same](https://almalinux.pkgs.org/8/almalinux-appstream-x86_64/copy-jdk-configs-4.0-2.el8.noarch.rpm.html) | 3.3 | 4.0 | | | |
| java-1.7.0-openjdk | | | | | | |
| java-1.7.0-openjdk-headless | | | | | | |
| javapackages-tools | | | | | | |
| jpackage-utils | | | | | | |
| pkgconfig | | | | | | |
| python-javapackages | | | | | | |

**Comment by h-vetinari** [comment]
@conda-forge/core, checking in here again - could people with opinions propose some ✅ / ❌ in the "keep as CDT?" column? (perhaps with a little abbreviation for your name, e.g. "Alice: ❌" or "Bob: ✅")

Especially the ❌'s would be important, but also stuff we absolutely _cannot_ move away from CDTs.

> x11 can probably fully dropped.

I mean we're all waiting for wayland to fully take over (right? 😉), though I guess you don't mean that we won't need x11 anymore (as a random example, there are >30 [recipes](https://github.com/search?q=org%3Aconda-forge+%2Fcdt%5C%28%5B%5C%22%5C%27%5Dlibx11%2F+path%3Arecipe%2Fmeta.yaml&type=code) still using the `libx11` CDT...), but rather that we should be able to move to [`xorg-libx11`](https://github.com/conda-forge/xorg-libx11-feedstock) etc.?

**Comment by kkraus14** [comment]
With the `rdma-core` feedstock (https://github.com/conda-forge/rdma-core-feedstock) most of the hardware / kernel interfaces are covered. Filling in the table now.

**Comment by isuruf** [comment]
> This [query](https://github.com/search?q=org%3Aconda-forge+%2Fcdt%5C%28%5B%5C%22%5C%27%5D%28expat%7Cpcre%7Clib%28jpeg-turbo%7Cpng%7Ctiff%7Cuuid%7Cxml2%29%29%28-devel%29%3F%2F+path%3Arecipe%2Fmeta.yaml&type=code) shows 26 recipes which use at least one of these directly, mostly expat. Another 8 feedstocks [have](https://github.com/search?q=org%3Aconda-forge+%2F%28expat%7Cpcre%7Clib%28jpeg-turbo%7Cpng%7Ctiff%7Cuuid%7Cxml2%29%29%28-devel%29%3F%2F+path%3Arecipe%2Fyum_requirements.txt&type=code) at least one of them in yum_requirements.txt.

It's not immediately clear if they are direct or indirect because some CDTs don't have dependencies correctly and the maintainers have to add them to the recipe even though they are indirect.

**Comment by h-vetinari** [comment]
Thanks a lot @kkraus14!

@isuruf I guess the simplest way to find out is to submit PRs which replace these cdts with our packages?

**Comment by h-vetinari** [comment]
Speaking about HW interfaces, we can probably also drop `alsa-lib` for https://github.com/conda-forge/alsa-lib-feedstock/.

Also, I'm pretty sure everything under "Fonts & Language" could be replaced (perhaps with the exception of `libthai` that we don't have yet)

**Comment by jakirkham** [comment]
Added a few more. Not positive whether we can drop the associated CDTs. So added question marks for now

**Comment by h-vetinari** [comment]
Tentatively marked the following as ❌ (principally because it's stuff we have feedstocks for, which should just work 🤞):
* `gdk-pixbuf2`, `glib-networking`, `glib2`, `gtk2`
* `cairo`, `libdrm`, `pixman`
* `alsa-lib`, `libnl`
* `kernel-headers`
* Everything under Fonts (except `libthai`)
* Everything under Web that has a clear feedstock
* `pkgconfig` (should be possible to use our own I believe)

Plus added some more feedstock matches, resp. candidates. Seems we have pretty much all of X11 already built separately (principally by @pkgw it seems 🙏)

Some near matches:
* Not sure how much of the mesa-* CDTs is covered by our [`mesalib`](https://github.com/conda-forge/mesalib-feedstock).
* I'm guessing our [`at-spi2-core`](https://github.com/conda-forge/at-spi2-core-feedstock) and [`at-spi2-atk`](https://github.com/conda-forge/at-spi2-atk-feedstock) cover most of what gtk/gnome should need
* our [`libnl`](https://github.com/conda-forge/libnl-feedstock) seems to cover at least v1 of the netlib interfaces; not sure about `libnl3` though (CC @nehaljwani)

**Comment by jakirkham** [comment]
The conda-forge `libnl` is [on version 3]( https://github.com/conda-forge/libnl-feedstock/blob/ce301674066b454b87180ed70ad2ab9b6cf50d37/recipe/meta.yaml#L2 ). So I think yes

Mesa is complicated. Maybe we can look at `gstreamer`, `opencv` or `qt-main` to test removing the associated CDTs

**Comment by h-vetinari** [comment]
This was again discussed in the core call today:
* @isuruf pointed out that we should be able to build `libglvnd`
* we can probably use X11 stack through the feedstocks for this
* security-critical libraries should almost certainly stay CDTs (likely `libselinux`, `libsepol`)
  * In this context we also discussed libpam (see #55), which should become a CDT for both cos7 & cos8.

Overall, we're getting to the point where we've come to a decision (if preliminary) for ~80% of CDTs, so we should be able to restart working on the alma enablement (which is painful due to changed repo structure / urls), rebuild perhaps slightly more CDTs than necessary, and then try to move feedstocks over to using our own builds, and only falling back to CDTs if that runs into issues.

**Comment by jaimergp** [comment]
- xref https://github.com/conda-forge/staged-recipes/pull/25919

**Comment by jakirkham** [comment]
Based on discussion at the core meeting it sounds like we are ok with the current list of CDTs

So we should be good to start producing these

cc @beckermr (for vis)

**Comment by isuruf** [comment]
That's not true. Please see the meeting note.

**Comment by h-vetinari** [comment]
> That's not true. Please see the meeting notes.

In the call, we did discuss how we're ready to start building all CDTs that aren't descoped in the mega-list above. We can still reduce that number later on if we prefer. The only open question was the effort around `libglvnd`, which shouldn't change the overall calculus AFAICT (except that we can leave out yet more packages once that becomes available). Perhaps you could update the matrix above with "❌**" or something like that, where the asterisks indicate that we can drop them once `libglvnd` becomes available?

**Comment by jakirkham** [comment]
Right it was my understanding we were done updating the list and ready to make CDTs

Could I ask that we spend the next week updating anything that still needs updates in the table?

Would like to get consensus for moving forward on this issue in the next core meeting. So we can start making these

**Comment by h-vetinari** [comment]
Thanks @isuruf for filling out the table! Is there something particular about `mesa-libgbm` that makes it stand out compared to all the other mesa libs, or was it just a simple editing oversight that it didn't get a "❌**"?

**Comment by h-vetinari** [comment]
Also, I've now marked all the X11-related packages for which we have feedstocks as ❌

**Comment by jakirkham** [comment]
In today's core meeting, we decided to go ahead with any that have a ✅

**Comment by jakirkham** [comment]
A user recently wrote up a list of X11 packages we are missing: https://github.com/conda-forge/staged-recipes/issues/26241

**Comment by ehfd** [comment]
https://github.com/conda-forge/libglvnd-feedstock is implemented. All OpenGL packages may be dropped completely.

However, as a substitute for Mesa CDTs, it is a good idea to implement Mesa with libglvnd as well.

**Comment by jakirkham** [comment]
> https://github.com/conda-forge/libglvnd-feedstock is implemented. All OpenGL packages may be dropped completely.

Thanks Seungmin! 🙏
 
> However, as a substitute for Mesa CDTs, it is a good idea to implement Mesa with libglvnd as well.

So would this mean updating [the `mesalib` feedstock]( https://github.com/conda-forge/mesalib-feedstock )? If so, that appears not to use `libglvnd` currently (from a CDT or otherwise)

**Comment by h-vetinari** [comment]
We have about [130 feedstocks](https://github.com/search?q=org%3Aconda-forge+%2F%28libgl%7Cmesa%29%2F+path%3Arecipe%2Fyum_requirements.txt&type=code) that use a yum-requirement that's either `mesa-*` or `libgl`-related. Sounds like we should maybe have a dedicated migrator (once an update mesa is in place?).

**Comment by ehfd** [comment]
libglvnd does not require mesa during package build, and this is a key point this feedstock exists. However, during runtime, if libglvnd doesn't find a vendor library, it will error out.
If there is a GPU and its driver, it will not be a problem. But it is a potential issue without any GPUs existing.
Either way, the feedstock is always better than the CDTs put in build dependencies.

**Comment by isuruf** [comment]
We don't need mesa in most packages that use mesa CDTs. Just replace with `libgl-devel` or `libegl-devel`.

**Comment by jakirkham** [comment]
> Just replace with `libgl-devel` or `libegl-devel`.

Initial attempt at such a change happening in PR: https://github.com/conda-forge/qt-main-feedstock/pull/284

**Comment by ehfd** [comment]
- [ ] xorg-libxv (https://gitlab.freedesktop.org/xorg/lib/libxv)
- [ ] xorg-libxxf86dga1 (https://gitlab.freedesktop.org/xorg/lib/libxxf86dga)
- [ ] xorg-libxshmfence (https://gitlab.freedesktop.org/xorg/lib/libxshmfence)

More info: https://github.com/conda-forge/staged-recipes/issues/26241

If someone could add the feedstocks (I lack time after spending them all for libglvnd) in the earliest future, I think the whole OpenGL/X11 CDT saga will be completed once and for all.

Especially, I see that many feedstocks use CDTs and `yum_requirements.txt` for libxshmfence and libxxf86dga1. These should be ported to feedstocks ASAP.

**Comment by h-vetinari** [comment]
> Especially, I see that many feedstocks use CDTs and `yum_requirements.txt` for libxshmfence and libxxf86dga1

I cannot find a single feedstock in conda-forge that [uses](https://github.com/search?q=org%3Aconda-forge+%2F.*xf86dga.*%2F+path%3Arecipe%2F&type=code) `libxxf86dga1` - which ones are you talking about? `libxshmfence` at least occurs in [4 feedstocks](https://github.com/search?q=org%3Aconda-forge+%2F.*shmfence.*%2F+path%3Arecipe%2F&type=code) - that does sound worthwhile to add.

**Comment by ehfd** [comment]
Looks like it is mostly only for mplayer, xdpyinfo (xorg-x11-utils), etc. Which do not have feedstocks.
@h-vetinari

**Comment by ehfd** [comment]
While libxv doesn't have used feedstocks because of no CDTs, it is something that can be enabled for better performance on many X11 aspects for GStreamer, potentially QT, etc. through build flags.

**Comment by h-vetinari** [comment]
First batch of builds landed in #70! 🥳

**Comment by h-vetinari** [comment]
Interested parties here can work towards removing usage of CDTs that we haven't lifted to alma8 (and `yum_requirements` that go along with it, especially those with a different name in alma8) across recipes.

We discussed in the core-call today that we'll probably do a separate migrator for that as well, but the less it has to chew through, the better. :)

Here's some searches to get you started (lots of X11, mesa, alsa etc. to be removed in both cases):
* [use of `cdt(...)`](https://github.com/search?q=org%3Aconda-forge+%2Fcdt%5C%28%2F+path%3Arecipe%2Fmeta.yaml&type=code)
* [use of `yum_requirements`](https://github.com/search?q=org%3Aconda-forge+path%3Arecipe%2Fyum_requirements.txt&type=code)

The only CDTs that can stay (if the feedstock still requires it) should be the following:
https://github.com/conda-forge/cdt-builds/blob/d020455bc70aa66905a81b63a5f11c566ca8f674/cdt_slugs.yaml#L1-L21

**Comment by hmaarrfk** [comment]
I think there is a question about the CDT builds as part of the tk + freetype.
https://github.com/conda-forge/tk-feedstock/pull/63

Without them we would have circular dependencies.

**Comment by h-vetinari** [comment]
I think this issue can be closed now?

**Comment by jakirkham** [comment]
Yep let's close this. We can always follow up on additional CDTs in new issues
